Compilation error for arch `wasm32` as of version 0.3.17

Hi there, thanks for the cool macros!

Unfortunately, the latest release broke something for me: starting with version 0.3.17, I get compilation errors for architecture `wasm32`. In particular, if I try to compile something like the minimal failing example you can find below, the compiler complains that it cannot find the function `terminal_size_of` of the crate `terminal_size`.

The function in question is feature-gated behind both `#[cfg(unix)]` and `#[cfg(windows)]`, but there's nothing for `#[cfg(wasm)]`. This probably makes sense, since `wasm` doesn't have a terminal (as far as I know). However, it would be sad if `assert2` required a terminal now where it didn't in the past. As far as I understand, the information gathered this way is only used for pretty-printing, and is not essential to the functionality of `assert2`.


## Minimal Failing Example


```toml
# Cargo.toml
[package]
name = "tmp"
version = "0.1.0"
edition = "2024"

[dev-dependencies]
assert2 = "=0.3.17"

[target.'cfg(target_arch = "wasm32")'.dev-dependencies]
wasm-bindgen-test = "0.3.42"
```

```rust
// lib.rs
#[cfg(test)]
use assert2::assert;

#[test]
#[cfg_attr(target_arch = "wasm32", wasm_bindgen_test::wasm_bindgen_test)]
fn it_works() {
    assert!(1 + 1 == 2);
}
```

## Detailed Error Output

```sh
   Compiling assert2 v0.3.17
error[E0425]: cannot find function `terminal_size_of` in crate `terminal_size`
  --> /…/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/assert2-0.3.17/src/__assert2_impl/print/mod.rs:69:51
   |
69 |         let (term_width, _term_height) = terminal_size::terminal_size_of(std::io::stderr())
   |                                                         ^^^^^^^^^^^^^^^^ help: a function with a similar name exists: `terminal_size`
   |
  ::: /…/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/terminal_size-0.4.3/src/lib.rs:39:1
   |
39 | pub fn terminal_size() -> Option<(Width, Height)> {
   | ------------------------------------------------- similarly named function `terminal_size` defined here
   |
note: found an item that was configured out
  --> /…/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/terminal_size-0.4.3/src/lib.rs:30:38
   |
28 | #[cfg(unix)]
   |       ---- the item is gated here
29 | #[allow(deprecated)]
30 | pub use crate::unix::{terminal_size, terminal_size_of, terminal_size_using_fd};
   |                                      ^^^^^^^^^^^^^^^^
note: found an item that was configured out
  --> /…/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/terminal_size-0.4.3/src/lib.rs:36:41
   |
34 | #[cfg(windows)]
   |       ------- the item is gated here
35 | #[allow(deprecated)]
36 | pub use crate::windows::{terminal_size, terminal_size_of, terminal_size_using_handle};
   |                                         ^^^^^^^^^^^^^^^^

For more information about this error, try `rustc --explain E0425`.
error: could not compile `assert2` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
Error: Compilation of your program failed
Caused by: Compilation of your program failed
Caused by: failed to execute `cargo build`: exited with exit status: 101
  full command: cd "/…/tmp" && CARGO_BUILD_TARGET="wasm32-unknown-unknown" "cargo" "build" "--tests"
```

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -18,7 +18,24 @@ jobs:
         run: |
           rustup install ${{ matrix.rust }} --profile minimal
           rustup default ${{ matrix.rust }}
-      - name: Build
-        run: cargo build --release --all-features --color=always
-      - name: Test
-        run: cargo test --release --all-features --color=always
+      - name: cargo build
+        run: cargo build --release --all-features --all-targets --color=always
+      - name: cargo test
+        run: cargo test --release --all-features --all-targets --color=always
+  check:
+    name:  check-only ${{ matrix.target }}, rust-${{ matrix.rust }}
+    strategy:
+      matrix:
+        target: [wasm32-unknown-unknown]
+        rust: [stable, nightly, 1.71.0]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@master
+      - name: Install Rust ${{ matrix.rust }} target ${{ matrix.target }}
+        run: |
+          rustup install ${{ matrix.rust }} --profile minimal
+          rustup default ${{ matrix.rust }}
+          rustup target add ${{ matrix.target }}
+      - name: cargo check
+        run: cargo check --release --all-features --all-targets --color=always --target ${{ matrix.target }}

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,6 @@
+main:
+  * Fix compilation on non-Unix and non-Windows platforms.
+
 v0.3.17 - 2026-02-15:
   * Support `let` chains in assertions.
   * Split top-level `&&` expressions, and highlight which one failed.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -19,6 +19,8 @@ assert2-macros = { version = "=0.3.17", path = "assert2-macros" }
 yansi = "1.0.1"
 diff = "0.1.13"
 unicode-width = "0.2.1"
+
+[target.'cfg(any(unix, windows))'.dependencies]
 terminal_size = "0.4.3"
 
 [dev-dependencies]

--- a/src/__assert2_impl/print/mod.rs
+++ b/src/__assert2_impl/print/mod.rs
@@ -61,14 +61,26 @@ pub enum Expansion<'a> {
 	Bool,
 }
 
+fn terminal_width() -> usize {
+	#[cfg(any(unix, windows))]
+	{
+		use terminal_size::{Width, terminal_size_of};
+		if let Some((Width(width), _height)) = terminal_size_of(std::io::stderr()) {
+			return width as usize;
+		}
+	}
+
+	// TODO: Is this fallback a good idea?
+	// Or is it better to disable features that misbehave when the terminal width is unknown?
+	80
+}
+
 impl<'a> FailedCheck<'a> {
 	#[rustfmt::skip]
 	pub fn print(&self) {
 		let mut buffer = String::new();
 		let options = options::AssertOptions::get();
-		let (term_width, _term_height) = terminal_size::terminal_size_of(std::io::stderr())
-			.map(|(w, h)| (w.0 as usize, h.0 as usize))
-			.unwrap_or((80, 80));
+		let term_width = terminal_width();
 		let mut writer = writer::WrappingWriter::new(&mut buffer, term_width, options.color);
 		self.print_assertion(&mut writer);
 		if !self.fragments.is_empty() {

